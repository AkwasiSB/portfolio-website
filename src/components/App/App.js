import React from 'react';
import Header from '../Header/Header';
import Home from '../Home/Home';
import About from '../About/About';
import Portfolio from '../Portfolio/Portfolio';
import Footer from '../Footer/Footer';
import styles from './App.scss';

export default class App extends React.Component {
	state = {
		isHeaderSticky: false,
		activeNavLink: 'Home',
	}

	componentDidMount() {
		window.addEventListener('scroll', this.handleWindowScroll);
	}

	componentWillUnmount() {
		window.removeEventListener('scroll', this.handleWindowScroll);
	}

	handleWindowScroll = () => {
		const header = document.getElementById('navbar');
		const sticky = header.offsetTop;
		let homeHeight = document.getElementById('home').offsetHeight;
		let aboutHeight = document.getElementById('about').offsetHeight;
		let portfolioHeight = document.getElementById('portfolio').offsetHeight;

		homeHeight += 60;
		aboutHeight += homeHeight;
		portfolioHeight += aboutHeight;

		if (window.pageYOffset < homeHeight) {
			this.setState({
				activeNavLink: 'Home',
				isHeaderSticky: window.pageYOffset > sticky ? true : false
			});
		}
		else if (window.pageYOffset >= homeHeight && window.pageYOffset < aboutHeight) {
			this.setState({
				activeNavLink: 'About',
				isHeaderSticky: window.pageYOffset > sticky ? true : false
			});
		}
		else if (window.pageYOffset >= aboutHeight && window.pageYOffset <= portfolioHeight) {
			this.setState({
				activeNavLink: 'Portfolio',
				isHeaderSticky: window.pageYOffset > sticky ? true : false
			});
		}
	}

	render() {
		const { isHeaderSticky, activeNavLink } = this.state;

		return (
			<div className={styles.mainContainer}>
				<Header isSticky={isHeaderSticky} activeNavLink={activeNavLink} />

				<main className={`${styles.content} ${isHeaderSticky ? 'paddTop' : ''}`}>
					<Home isHeaderSticky={isHeaderSticky} />
					<About />
					<Portfolio />
				</main>

				<Footer />
			</div>
		);
	}
}
