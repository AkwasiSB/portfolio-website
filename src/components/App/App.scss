.main-container {
	display: grid;
	grid-template-rows: [row-line1] 6rem [row-line2] 8fr [row-line3] 6rem [row-line4];
	grid-template-areas:
		"navbar"
		"content"
		"footer";
	align-content: stretch;
	background-image: linear-gradient(to bottom, #eee , #fff);
}

:local(.content) {
	grid-area: content;

	:global &.paddTop {
		padding-top: 0;
	}
}
