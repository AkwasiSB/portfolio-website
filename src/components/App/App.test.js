import App from './App';

test('should render <App /> correctly', () => {
  const appWrapper = shallow(<App />);

  expect(appWrapper).toMatchSnapshot();
})
