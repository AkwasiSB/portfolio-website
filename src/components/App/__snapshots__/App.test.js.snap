// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <App /> correctly 1`] = `
<div
  className="mainContainer"
>
  <Header
    activeNavLink="Home"
    isSticky={false}
  />
  <main
    className="content "
  >
    <Home
      isHeaderSticky={false}
    />
    <About />
    <Portfolio />
  </main>
  <Footer />
</div>
`;
